packages/core/test/request-builder/v4/get-all-request-builder.spec.ts

```diff
@@ -72,6 +80,58 @@ describe('GetAllRequestBuilderV4', () => {
       ]);
     });
 
+    it('type tests', async () => {
+      // TestEntityTemporal<string, number>[]
+      const res1 = await TestEntityTemporal.requestBuilder()
+        .getAll()
+        .execute(defaultDestination);
+      // TestEntityTemporal<string, number>[]
+      const res2 = await TestEntityTemporal.requestBuilder(defaultMiddleware)
+        .getAll()
+        .execute(defaultDestination);
+      // TestEntityTemporal<string, string>[]
+      const res3 = await TestEntityTemporal.requestBuilder(customMiddlewareFull)
+        .getAll()
+        .execute(defaultDestination);
+      // TestEntityTemporal<string, string>[]
+      const res4 = await TestEntityTemporal.requestBuilder(
+        customMiddlewarePartial
+      )
+        .getAll()
+        .execute(defaultDestination);
+
+      // TestEntityTemporal<string, number>
+      const entity1 = TestEntityTemporal.builder().int32Property(1).build();
+      // TestEntityTemporal<string, string>
+      const entity2 = TestEntityTemporal.builder(customMiddlewareFull)
+        .int32Property('1')
+        .build();
```

Comment:
With/Without the middleware parameter, see the difference of `int32Property()` that accepts different types.